File: leetcode/7reverse_integer/_7reverse_integer.py
"""
7. Reverse Integer
Have you thought about this?
Here are some good questions to ask before coding. 
Bonus points for you if you have already thought through this!
If the integer's last digit is 0, what should the output
 be? ie, cases such as 10, 100.
Did you notice that the reversed integer might overflow?
 Assume the input is a 32-bit integer, then the reverse 
 of 1000000003 overflows. How should you handle such cases?
For the purpose of this problem, assume that your function 
returns 0 when the reversed integer overflows.
"""
while True :
    x=int(input())
    a = str(x)
    b =''
    c=0
    if a[0]=='-':
        b=a[::-1]
        b=b[0:len(b)-1]
        c=-int(b)
    else :
        b=a[::-1]
        c=int(b)
    if c>2147483648 or c<=-2147483648:
        print(0)
    print(c)

File: leetcode/7reverse_integer/7reverse_integer.c
/*
7. Reverse Integer 
Example1: x = 123, return 321
Example2: x = -123, return -321
*/
#include<stdio.h>
int reverse(int x) {
	if (x > 2147483648 || x <= -2147483648)
		return 0;
	return 0;
}
int main() {
	int x=-12;
	int j = 1,a[20];
	if (x < 0) {
		x = -x;
		while (x >= 10) {
			x= x / 10;
			j++;
		}
		printf("%d", j);
	}
	else {
		while (x >= 10) {
			x = x / 10;
			j++;
		}
		printf("%d", j);
	}
}
